### ISSUE TITLE
-walletnotify on 6 confirmations instead of 1?

### ISSUE THREAD

**Issue by cryptoapi** [bitcoin/bitcoin#10021]
We need the same functionality, for our Gourl.io Payment gateway, as described here - 

http://bitcoin.stackexchange.com/questions/40689/walletnotify-on-6-confirmations-instead-of-1

----
Pretty straight forward. Right now im using -walletnotify to run some node.js code when i get an incoming transaction and when theres 1 confirmation. However, after reading around, i found out that 1 confirmation is not secure, which now makes me want to execute this confirmation code when theres at least 6 confirmations.

TL;DR, Is there a parameter or an equivalent for -walletnotify to run on N confirmations?
------------

**Comment by laanwj** [comment]
> TL;DR, Is there a parameter or an equivalent for -walletnotify to run on N confirmations?

There is no "a transaction reached N confirmations" event, nor any internal infrastructure to handle such.
You need to do this manually at the moment. For example subscribe to `-blocknotify` and there query the set of transactions that has not yet reached your desired number of confirmations.

**Comment by cryptoapi** [comment]
Can you add internal infrastructure to handle such in next releases ? 
It is very useful functionality for many e-commerce companies who accept bitcoin as payment.

**Comment by laanwj** [comment]
> Can you add internal infrastructure to handle such in next releases ?

I personally? No, I don't have time to work on this.
If you really need this you'll have to make sure it gets implemented yourself. Of course I'm willing to help/assist where needed.

**Comment by jonasschnelli** [comment]
You can do that (in a non elegant way) be adding a ZMQ listener for new blocks, then RPC `listtransactions` and then check your transactions confirmation count.

> Can you add internal infrastructure to handle such in next releases ?

This is an open source project. If your company depends on this project you should probably consider participating in some form.

**Comment by MarcoFalke** [comment]
The feature request didn't seem to attract much attention in the past. Also, the issue seems not important enough right now to keep it sitting around idle in the list of open issues.

Closing due to lack of interest. Pull requests with improvements are always welcome.
